### PR TITLE
For profiling add symbolic links so Igprof and Circles files can be read

### PR DESCRIPTION
**Pull request by gartung**
For profiling test results it would be useful to have Igprof and Circles available to browse. This is done with relative symbolic links in the igprof and profiling directories for the nightly IB release built against. 

ToDo: How to make links to the files?

### CODE REVIEW

**Comment by gartung** [comment]
enable profiling

**Comment by gartung** [comment]
test

File: pr_testing/test_multiple_prs.sh [hunk 1]
```diff
@@ -1181,23 +1181,37 @@ for BT in ${ENABLE_BOT_TESTS}; do
...
+                     $WORKSPACE/upload/igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/$(basename $f)
              done
-             for f in $(find $PROFILING_WORKFLOW -type f -name '*.sql3' -o -name '*.log' -o -name '*.json' -o -name '*.txt') ; do
+             for f in $(find $PROFILING_WORKFLOW -type f -name '*.json' ) ; do
+               d=$(dirname $f)
+               mkdir -p $WORKSPACE/upload/profiling/$d || true
+               cp -p $f $WORKSPACE/upload/profiling/$d/ || true
+               mkdir -p $WORKSPACE/upload/profiles/$d || true
+               mkdir -p $WORKSPACE/upload/profiling/${CMSSW_IB}/${ARCHITECTURE}/${PROFILING_WORKFLOW}/PR-${REPORT_H_CODE}/${BUILD_NUMBER}
+               ln -s ../../../../../pull-request-integration/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/profiling/$d/$(basename $f)  \
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:1203]: @gartung , should not you be creating symlink pointing to 
`igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/` ? Idea was to create a symlink in pull-request-integration directory for the release. So for each PR tests there will be a symlink which points to the release igprof profiling

Comment (gartung) [reply pr_testing/test_multiple_prs.sh:1203]: The link name is on the next line after the continuation \

Comment (smuzaffar) [reply pr_testing/test_multiple_prs.sh]: but here you are creating a file (symlink) which actually points to `../../../../../pull-request-integration/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/profiling/$d/$(basename $f)`  ,. I am saying that it should point to `../../../../../igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/`

Comment (smuzaffar) [reply pr_testing/test_multiple_prs.sh]: PR artifects are uploaded uploaded in to `pull-request-integration/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/` directory and you want to create a symlink pointing  to the files of the release area

**Comment by gartung** [comment]
I want to link this
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14ac78/9288/profiling/23434.21/
under this
https://cmssdt.cern.ch/SDT/jenkins-artifacts/igprof/CMSSW_11_2_X_2020-09-14-1100/slc7_amd64_gcc820/profiling/23434.21/
and this
https://cmssdt.cern.ch/SDT/jenkins-artifacts/profiling/CMSSW_11_2_X_2020-09-10-1100/slc7_amd64_gcc820/23434.21/

File: pr_testing/test_multiple_prs.sh (hunk 1 above)
Comment (gartung) [reply pr_testing/test_multiple_prs.sh]: The sql files need to linked under jenkins_artifcats/igprof/ and the json files need to be linked under jenkins_artifacts/profiling to be browsable with igprof_navigator and circles.

Comment (smuzaffar) [reply pr_testing/test_multiple_prs.sh]: ah right.... I got the point. Yes currently igprof_navigator has a hardcoded path.

**Comment by gartung** [comment]
@smuzaffar The links did not get uploaded.
https://cmssdt.cern.ch/jenkins/job/ib-run-pr-tests/9338/console

**Comment by smuzaffar** [comment]
ah thisis because we are uploading the contents under pull-request-integration/PR-${REPORT_H_CODE}/${BUILD_NUMBER} ( https://github.com/cms-sw/cms-bot/blob/559aa60e761696ed3995025298a5d6e38dc1bbc9/pr_testing/test_multiple_prs.sh#L1278-L1283 ). I would suggest to create the symlink under $WORKSPACE/profile and add a section like https://github.com/cms-sw/cms-bot/blob/559aa60e761696ed3995025298a5d6e38dc1bbc9/pr_testing/test_multiple_prs.sh#L1280-L1282 to upload it under release profile directory

File: pr_testing/test_multiple_prs.sh [hunk 2]
```diff
@@ -1303,6 +1301,12 @@ if [ -z ${NO_POST} ] ; then
     if [ -d $LOCALRT/das_query ] ; then
       send_jenkins_artifacts $LOCALRT/das_query das_query/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/PR || true
     fi
+    if [ -d $LOCALRT/profiling ]; then
+      send_jenkins_artifacts $LOCALRT/profiling/${CMSSW_IB}/${ARCHITECTURE}/${PROFILING_WORKFLOW}/ profiling/${CMSSW_IB}/${ARCHITECTURE}/${PROFILING_WORKFLOW}/
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:1305]: @gartung , what will happen when more than one PR run profiling or pr tests run multiple times for the same PR?

Comment (gartung) [reply pr_testing/test_multiple_prs.sh]: Updated.

File: pr_testing/test_multiple_prs.sh [hunk 3]
```diff
@@ -1280,6 +1301,16 @@ if [ -z ${NO_POST} ] ; then
     if [ -d $LOCALRT/das_query ] ; then
       send_jenkins_artifacts $LOCALRT/das_query das_query/PR-${REPORT_H_CODE}/${BUILD_NUMBER}/PR || true
     fi
+    if [ -d $LOCALRT/profiling ]; then
+      for PROFILING_WORKFLOW in $PROFILING_WORKFLOWS;do
+        send_jenkins_artifacts $LOCALRT/profiling/${CMSSW_IB}/${ARCHITECTURE}/${PROFILING_WORKFLOW} profiling/${CMSSW_IB}/${ARCHITECTURE}/${PROFILING_WORKFLOW}/
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:1306]: why not just upload `$LOCALRT/profiling/${CMSSW_IB}/${ARCHITECTURE}` to `profiling/${CMSSW_IB}/${ARCHITECTURE}` ? Do you have any thing else under  $LOCALRT/profiling/${CMSSW_IB}/${ARCHITECTURE} ?

File: pr_testing/test_multiple_prs.sh [hunk 4]
```diff
@@ -1280,6 +1301,16 @@ if [ -z ${NO_POST} ] ; then
...
+      done
+    fi
+    if [ -d $LOCALRT/igprof ]; then
+      for PROFILING_WORKFLOW in $PROFILING_WORKFLOWS;do
+        send_jenkins_artifacts $LOCALRT/igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW} igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:1311]: same here

**Comment by gartung** [comment]
@smuzaffar the links for igprof-navigator are in the directory
https://cmssdt.cern.ch/SDT/jenkins-artifacts/igprof/CMSSW_11_2_X_2020-09-16-1100/slc7_amd64_gcc820/profiling/23434.21/PR-744a42/9379/
I think I have the link correct here
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-744a42/9379/profiling/
https://cmssdt.cern.ch/SDT/cgi-bin/igprof-navigator/CMSSW_11_2_X_2020-09-16-1100/slc7_amd64_gcc820/profiling/23434.21/PR-744a42/9379/igprofCPU_step4.sql3
but igprof-navigator is not showing anything. Does it only work with real files?

**Comment by smuzaffar** [comment]
@gartung , I do not remember but please checkhttps://github.com/cms-sw/cmssdt-web/blob/master/cgi-bin/igprof-navigator#L606  and see why it is not getting files from the PR

**Comment by gartung** [comment]
If I remove the filename from the link it gives me the directory listing.
https://cmssdt.cern.ch/SDT/cgi-bin/igprof-navigator/CMSSW_11_2_X_2020-09-16-1100/slc7_amd64_gcc820/profiling/23434.21/PR-744a42/9379/
and the files are clickable from there. I can update the links to just use the directory name.

**Comment by gartung** [comment]
Oh now I see I need to drop the .sql extension.

**Comment by gartung** [comment]
@smuzaffar This is ready to go.

**Pull request merged**